src/ui/common/src/components/operators/RequireOperator.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { OperatorResultResponse } from '../../handlers/responses/operator';
+
+type Props = {
+  operator?: OperatorResultResponse;
+  children?: React.ReactElement | React.ReactElement[];
+};
+
+const RequireOperator: React.FC<Props> = ({ operator, children }) => {
+  const navigate = useNavigate();
+  if (!operator) {
+    navigate('/404');
```

Comment:
I wonder if we can pass a message here saying that we cannot find the operator / artifact specified?

That way users have a little more context into why they are seeing a 404 page.